tests/basic/volume_ops_test.go

```diff
@@ -3202,6 +3202,120 @@ var _ = Describe("{FioClonedVolumeFaultInjection}", func() {
 		}
 	})
 })
+var _ = Describe("{VolumePreCheck}", func() {
+	/*
+			https://portworx.atlassian.net/browse/PTX-20557
+			1. Deploy a basic volume on the cluster
+			2. Now we need to test the ha-update of the volume with different cases
+			scenarios :
+		    1. Test the ha-update sources option with an uuid of other nodes on which node is not present
+		    2. Test the ha-update sources option with an ip of nodes
+			3. Test the ha-update sources option with an invalid uuid of the node on which the repl of the volume is present
+		    4. Test the ha-update sources option with a valid uuid of the node on which the repl of the volume is present
+
+	*/
+	JustBeforeEach(func() {
+		StartTorpedoTest("volumeprecheck", "Precheck for volume operations", nil, 0)
+	})
+
+	It("volumeprecheck", func() {
+		log.InfoD("volumeprecheck")
+		stepLog := "Create a volume and perform a precheck for sources options"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+
+			stNodes := node.GetStorageDriverNodes()
+			nodesuuidWithoutReplica := make([]string, 0)
+			nodesIP := make([]string, 0)
+
+			index := rand.Intn(len(stNodes))
+			selectedNode := &stNodes[index]
+
+			var aggr_level int
+			var repl_level int
+			storageNodes := node.GetStorageNodes()
+			if len(storageNodes) >= 4 && len(storageNodes) < 9 {
+				log.InfoD("Setting the aggr_level and repl_level to 2 as storage nodes in the cluster are %d", len(storageNodes))
```

Comment:
nit: you can print this log at once after setting aggr_level and repl_level values